src/ui/pages/Networks/Networks.tsx

```diff
@@ -38,119 +42,80 @@ import { UnstyledLink } from 'src/ui/ui-kit/UnstyledLink';
 import { VStack } from 'src/ui/ui-kit/VStack';
-import { ViewSuspense } from 'src/ui/components/ViewSuspense';
 import { useDebouncedCallback } from 'src/ui/shared/useDebouncedCallback';
 import { SearchInput } from 'src/ui/ui-kit/Input/SearchInput';
 import { BottomSheetDialog } from 'src/ui/ui-kit/ModalDialogs/BottomSheetDialog';
 import type { HTMLDialogElementInterface } from 'src/ui/ui-kit/ModalDialogs/HTMLDialogElementInterface';
 import { HStack } from 'src/ui/ui-kit/HStack';
 import { UIText } from 'src/ui/ui-kit/UIText';
 import { showConfirmDialog } from 'src/ui/ui-kit/ModalDialogs/showConfirmDialog';
-import { SurfaceItemLink, SurfaceList } from 'src/ui/ui-kit/SurfaceList';
-import { NetworkForm } from './NetworkForm';
-import { NetworkList } from './shared/NetworkList';
-import { SearchResults } from './shared/SearchResults';
-import { LocationStateHelperStore } from './shared/LocationStateHelperStore';
-import { createEmptyNetwork } from './shared/createEmptyNetwork';
+import type { ChainDistribution } from 'src/ui/shared/requests/PortfolioValue/ChainValue';
+import type { NetworkGroups } from 'src/ui/components/NetworkSelectDialog/createNetworkGroups';
+import { createGroups } from 'src/ui/components/NetworkSelectDialog/createNetworkGroups';
+import { AddNetworkLink } from 'src/ui/components/NetworkSelectDialog/AddNetworkLink';
+import { useSearchKeyboardNavigation } from 'src/ui/components/NetworkSelectDialog/useSearchKeyboardNavigation';
+import { LIST_ITEM_CLASS } from 'src/ui/components/NetworkSelectDialog/constants';
+import { KeyboardShortcut } from 'src/ui/components/KeyboardShortcut';
+import { isCustomNetworkId } from 'src/modules/ethereum/chains/helpers';
+import type { AddEthereumChainParameter } from 'src/modules/ethereum/types/AddEthereumChainParameter';
+import { toAddEthereumChainParameter } from 'src/modules/networks/helpers';
+import { usePreferences } from 'src/ui/features/preferences';
+import { useWalletAddresses } from './shared/useWalletAddresses';
 import { NetworkCreateSuccess } from './NetworkCreateSuccess';
+import { createEmptyChainConfig } from './shared/createEmptyChainConfig';
+import { SearchResults } from './shared/SearchResults';
+import { NetworkList } from './shared/NetworkList';
+import { NetworkForm } from './NetworkForm';
 
-async function saveNetworkConfig(network: NetworkConfig) {
-  const networks = await networksStore.load();
-  const ethereumChainConfig = networks.findEthereumChainById(
-    network.external_id
+async function saveChainConfig({
+  chain,
+  chainConfig,
+  prevChain,
+}: {
+  chain: string;
+  chainConfig: AddEthereumChainParameter;
+  prevChain?: string;
+}) {
+  const networks = await networksStore.load(
+    [chain, prevChain].filter(isTruthy)
   );
+  const chainsMetadata = networks.getNetworksMetaData();
+  const metadata = chainsMetadata[prevChain || chain];
+  if (prevChain && prevChain !== chain) {
+    await walletPort.request('updateChainForAffectedOrigins', {
```

Comment:
I don't think this method should be called from the ui side. It's an internal accidental cleanup